test/screenshot/spec/mdc-snackbar/classes/baseline-with-action.html

```diff
@@ -47,12 +47,11 @@
       <button class="test-snackbar-open-button test-font--redact-all" data-test-snackbar-id="test-snackbar--with-action" autofocus>Open snackbar with action</button>
 
       <div class="mdc-snackbar mdc-snackbar--open"
-           id="test-snackbar--with-action"
-           role="status"
-           aria-live="polite"
-           aria-hidden="true">
+           id="test-snackbar--with-action">
         <div class="mdc-snackbar__surface">
-          <div class="mdc-snackbar__label">Can't send photo. Retry in 5 seconds.</div>
+          <div class="mdc-snackbar__label"
+               role="status"
+               aria-live="polite">Can't send photo. Retry in 5 seconds.</div>
```

Comment:
Great question!

Yes, I think we want to keep it here in the markup so that clients can easily change it to `assertive` if necessary (some screen readers don't announce the text unless it's `assertive`).

I experimented with having `announce()` take `message` and `priority` args, but I wasn't able to make it work in practice. Screen readers are _really_ picky about where you put the ARIA attributes and do the DOM manipulation, and sometimes that needs to happen on different elements.

I also tried doing something like [`goog.a11y.aria.Announcer#getLiveRegion_()`](https://chromium.googlesource.com/chromium/src/third_party/+/master/google_input_tools/third_party/closure_library/closure/goog/a11y/aria/announcer.js?autodive=0%2F#101), dynamically inserting an anonymous element into the `<body>` using JS. But unfortunately some screen reader/browser combinations refuse to announce the text when I do that. Others needed a much bigger timeout value between clearing and resetting the text.

I've arrived at the present solution after many hours of painful experimentation. It seems to work reliably in 6 of the most commonly-used browser/screen reader combinations: JAWS and NVDA in Chrome, IE 11 💩, and Firefox.

It doesn't seem to work with Safari + VoiceOver on macOS 10.13 High Sierra, unfortunately.
I'll document that limitation in the README (next PR).

Reply:
In my testing it seemed to work with assertive rather than polite, but I can re-test when we get to the readme.

Reply:
Thanks for explaining! The announce adapter method seem to be coupled with what util.announce() method does which has its own business logic. It has to follow certain sequence of DOM operations to implement it.

For example, some wrappers have their own announcer utilities which makes it hard for them to reuse it or do they've to reimplement this util function?

Reply:
I expect internal teams to use [`goog.a11y.aria.Announcer#say()`](https://chromium.googlesource.com/chromium/src/third_party/+/master/google_input_tools/third_party/closure_library/closure/goog/a11y/aria/announcer.js?autodive=0%2F#80), and all other wrappers to use `util.announce()`.

Reply:
As discussed offline, as long as if the screen reader doesn't end up start pronouncing it twice (or jank) because of `aria-live` on child element and from announcer service, I guess this works for everyone.